software/contrib/piconacci.py

```diff
@@ -0,0 +1,145 @@
+from europi import *
+import machine
+from time import ticks_diff, ticks_ms
+from random import randint, uniform, choice
+
+from europi_script import EuroPiScript
+
+"""
+Piconacci
+author: Sean Bechhofer (github.com/seanbechhofer)
+date: 2022-05
+labels: triggers, maths
+
+Triggers based on the Fibonacci sequence
+
+digital_in: clock in
+analog_in: 
+
+knob_1: 
+knob_2: 
+
+button_1: Short Press: Move window on sequence left. Long Press: Rotate values left
+button_2: Short Press: Move window on sequence to the right. Long Press: Rotate values right
+output_1: trigger
+output_2: trigger
+output_3: trigger
+output_4: trigger
+output_5: trigger
+output_6: trigger
+
+"""
+
+
+# This probably works for most sensible circumstances. The 50th value
+# (disregarding 0, 1) is 20,365,011,074. At a BPM of 120, that'll be a
+# trigger every 300 years which is quite a lot even for slow moving
+# ambient pieces,
+MAX_FIB = 50
+
+
+class Piconacci(EuroPiScript):
+    def __init__(self):
+        # Flag to indicate whether the display needs updating.
+        self.dirty = True
```

Comment:
[optional] Consider variable names that make descriptive comments redundant, allowing you to remove or reduce them. For example instead of `dirty` you could call this something like `display_update_required`. It's a little more verbose, but the comments here and where it is used become unnecessary.